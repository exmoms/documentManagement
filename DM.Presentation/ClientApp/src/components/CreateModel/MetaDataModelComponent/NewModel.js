import React from "react";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import TextField from "@material-ui/core/TextField";
import Autocomplete from "@material-ui/lab/Autocomplete";
import { useTranslation } from "react-i18next";

function ComboBox(props) {
  // Convert Array of string to Array of Objects because "Autocomplete" component need array of object for "options" attribute,
  // otherwise gives errors.

  const { t } = useTranslation();

  return (
    <Autocomplete
      value={props.value.documentClassSelected}
      // defaultValue={props.value.documentClassSelected}
      onChange={props.onChange}
      id="combo-box-demo"
      options={props.options}
      getOptionLabel={(option) =>
        option.documentClassName !== window.undefined
          ? option.documentClassName
          : ""
      }
      renderInput={(params) => (
        <TextField
          error={
            props.value.documentClassSelected.id === window.undefined &&
            props.value.step_1_error === true
              ? true
              : false
          }
          helperText={
            props.value.documentClassSelected.id === window.undefined &&
            props.value.step_1_error === true
              ? t("fill_out_field")
              : ""
          }
          {...params}
          label={t("selectdocumentclass")}
          variant="outlined"
          fullWidth
        />
      )}
    />
  );
}

export default function AddressForm(props) {
  const onchange = (event, value) => {
    if (event.target.name === "modelName") {
      props.onSetModelName(event.target.value);
    }
  };

  const handelComboBox = (event, value) => {
    if (value) {
      props.onSetModelClassId(value.id);
      props.handeldocumentClassSelected(value);
    }
  };

  const { t } = useTranslation();
  return (
    <React.Fragment>
      <Typography variant="h6" gutterBottom align="left">
        {t("documentmodel")}
      </Typography>
      <Grid container spacing={3}>
        <Grid item xs={12} sm={6}>
          <TextField
            error={
              props.value.metaDataModelName === "" &&
              props.value.step_1_error === true
                ? true
                : false
            }
            helperText={
              props.value.metaDataModelName === "" &&
              props.value.step_1_error === true
                ? t("fill_out_field")
                : ""
            }
            required
            id="modelName"
            name="modelName"
            label={t("namelabel")}
            fullWidth
            autoComplete="off"
            value={props.value.metaDataModelName}
            onChange={onchange}
          />
        </Grid>
        <Grid item xs={12} sm={6}>
          <ComboBox
            onChange={handelComboBox}
            options={props.value.DocumentClassList}
            value={{
              documentClassSelected: props.value.documentClassSelected,
              step_1_error: props.value.step_1_error,
            }}
            className="ComboBox"
          />
        </Grid>
      </Grid>
    </React.Fragment>
  );
}
