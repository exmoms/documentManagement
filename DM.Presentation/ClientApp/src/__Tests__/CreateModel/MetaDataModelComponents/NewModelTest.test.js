import React from "react";
import ReactDom from "react-dom";
import NewModel from "../../../components/CreateModel/MetaDataModelComponent/NewModel.js";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import { act } from "react-dom/test-utils";
afterEach(cleanup);
describe("NewModel", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    const onSetModelName = jest.fn();
    const onSetModelClassId = jest.fn();
    const handelDocumentClassSelectedMock = jest.fn();
    let MetaDataModelNameMocked = "";
    let DocumentClassIdMocked = 0;
    let DocumentClassListMocked = [];
    let DocumentClassSelectedMocked = [{}];
    ReactDom.render(
      <NewModel
        onSetModelName={onSetModelName}
        onSetModelClassId={onSetModelClassId}
        handeldocumentClassSelected={handelDocumentClassSelectedMock}
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassId: DocumentClassIdMocked,
          DocumentClassList: DocumentClassListMocked,
          documentClassSelected: DocumentClassSelectedMocked,
        }}
      />,
      div
    );
  });
  test("matches snapshot", async () => {
    const onSetModelName = jest.fn();
    const onSetModelClassId = jest.fn();
    const handelDocumentClassSelectedMock = jest.fn();
    let MetaDataModelNameMocked = "";
    let DocumentClassIdMocked = 0;
    let DocumentClassListMocked = [];
    let DocumentClassSelectedMocked = [{}];
    const wrapper = shallow(
      <NewModel
        onSetModelName={onSetModelName}
        onSetModelClassId={onSetModelClassId}
        handeldocumentClassSelected={handelDocumentClassSelectedMock}
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassId: DocumentClassIdMocked,
          DocumentClassList: DocumentClassListMocked,
          documentClassSelected: DocumentClassSelectedMocked,
        }}
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("test passing data to props functionality", () => {
    const onSetModelNameMock = jest.fn();
    const onSetModelClassIdMock = jest.fn();
    const handelDocumentClassSelectedMock = jest.fn();
    let MetaDataModelNameMocked = "";
    let DocumentClassIdMocked = 0;
    let DocumentClassListMocked = [];
    let DocumentClassSelectedMocked = [{ documentClassId: 1 }];
    let wrapper;
    wrapper = shallow(
      <NewModel
        onSetModelName={onSetModelNameMock}
        onSetModelClassId={onSetModelClassIdMock}
        handeldocumentClassSelected={handelDocumentClassSelectedMock}
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassId: DocumentClassIdMocked,
          DocumentClassList: DocumentClassListMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          step_1_error:true
        }}
      />
    );
    //expectations
    expect(wrapper.find(TextField).at(0).props().value).toBe(
      MetaDataModelNameMocked
    );
    expect(wrapper.find(".ComboBox").at(0).props().value).toStrictEqual({"documentClassSelected": [{"documentClassId": 1}], "step_1_error": true});
    expect(wrapper.find(".ComboBox").at(0).props().options).toBe(
      DocumentClassListMocked
    );
    act(() => {
      wrapper
        .find(TextField)
        .at(0)
        .simulate("change", { target: { value: "test2", name: "modelName" } });
      wrapper.find(".ComboBox").at(0).simulate("change", "", { id: 1 });
    });
    expect(onSetModelNameMock).toHaveBeenCalledTimes(1);
    expect(onSetModelNameMock.mock.calls[0][0]).toStrictEqual("test2");
    expect(onSetModelClassIdMock).toHaveBeenCalledTimes(1);
    expect(onSetModelClassIdMock.mock.calls[0][0]).toStrictEqual(1);
    expect(handelDocumentClassSelectedMock).toHaveBeenCalledTimes(1);
    expect(handelDocumentClassSelectedMock.mock.calls[0][0]).toStrictEqual({
      id: 1,
    });
  });

  test("error and helperText props functionality", () => {
    const onSetModelNameMock = jest.fn();
    const onSetModelClassIdMock = jest.fn();
    const handelDocumentClassSelectedMock = jest.fn();
    let MetaDataModelNameMocked = "";
    let DocumentClassIdMocked = 0;
    let DocumentClassListMocked = [];
    let DocumentClassSelectedMocked = [{ documentClassId: 1 }];
    let wrapper;
    wrapper = shallow(
      <NewModel
        onSetModelName={onSetModelNameMock}
        onSetModelClassId={onSetModelClassIdMock}
        handeldocumentClassSelected={handelDocumentClassSelectedMock}
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassId: DocumentClassIdMocked,
          DocumentClassList: DocumentClassListMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          step_1_error:true
        }}
      />
    );
    //expectations
    expect(wrapper.find(TextField).at(0).props().error).toBeTruthy();
    expect(wrapper.find(TextField).at(0).props().helperText).toBe("fill_out_field");
  });

  test("error and helperText props functionality when false ", () => {
    const onSetModelNameMock = jest.fn();
    const onSetModelClassIdMock = jest.fn();
    const handelDocumentClassSelectedMock = jest.fn();
    let MetaDataModelNameMocked = "";
    let DocumentClassIdMocked = 0;
    let DocumentClassListMocked = [];
    let DocumentClassSelectedMocked = [{ documentClassId: 1 }];
    let wrapper;
    wrapper = shallow(
      <NewModel
        onSetModelName={onSetModelNameMock}
        onSetModelClassId={onSetModelClassIdMock}
        handeldocumentClassSelected={handelDocumentClassSelectedMock}
        value={{
          metaDataModelName: MetaDataModelNameMocked,
          documentClassId: DocumentClassIdMocked,
          DocumentClassList: DocumentClassListMocked,
          documentClassSelected: DocumentClassSelectedMocked,
          step_1_error:false
        }}
      />
    );
    //expectations
    expect(wrapper.find(TextField).at(0).props().error).toBeFalsy();
    expect(wrapper.find(TextField).at(0).props().helperText).toBe("");
  });
});
