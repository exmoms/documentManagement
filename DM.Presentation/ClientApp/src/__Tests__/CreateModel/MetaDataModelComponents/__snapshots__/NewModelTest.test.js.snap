// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewModel matches snapshot 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Typography))
    align="left"
    gutterBottom={true}
    variant="h6"
  >
    documentmodel
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <WithStyles(ForwardRef(TextField))
        autoComplete="off"
        error={false}
        fullWidth={true}
        helperText=""
        id="modelName"
        label="namelabel"
        name="modelName"
        onChange={[Function]}
        required={true}
        value=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      sm={6}
      xs={12}
    >
      <ComboBox
        className="ComboBox"
        onChange={[Function]}
        options={Array []}
        value={
          Object {
            "documentClassSelected": Array [
              Object {},
            ],
            "step_1_error": undefined,
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
